### DISCUSSION TITLE
Cómo puedo precargar un componente de tipo coleccion con un valor status (1,2,3,4,5) de la task o con una propiedad del canal?

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#375] [2. Questions and Answers] [answered]
Hola a todos

Necesito hacer lo siguiente:

1.- Tengo un formulario que debe estar en X estados (sesion 1, sesión 2, sesión 3, sesión 4......) para que se pueda utlizar. El tema que tengo es que, en el momento de levantar el formulario en la sesion 3, me deba aparecer en el campo el N°**3** y así sucesivamente hasta que el formulario no se siga utilizando. El problema que tengo es cómo lo puedo precargar dicho campo desde el status de la task, así se podría actualizar el valor y dejar registrado los valores.

2.- Tengo otra opción es hacer un preload relacionado al user, pero con una property en especifico, no sé si me serviria para property#channel, debido a que tiene como coleccion **propertyIds** y desconozco si se puede que se busque por propertyType

Adjunto un preload ejemplo

```
    "exec" : {
        "preload" : {
            "src" : "
function run () { 
  const commands = []; 
  const user = context['user#me']; 
  commands.push({ 
    cmd: 'SET_RESPONSES', 
    value: user.properties[0]
  });
  return commands;
}
        ", 
            "context" : "property#user,user#me"
        }
    }
```

Gracias =>

**Comment by acedward** [comment, marked as answer]
Sí, property#channel debería tener todas las propiedades de la tarea.
Podrías filtrarlo con

```
  const value = context['property#channel'].find(property => property.propertyType === 'Nombre-Tipo-Propiedad');
  commands.push({ 
    cmd: 'SET_RESPONSES', 
    value: value
  });
```

**Reply by NicolasPLG** [reply]
Lo probaré muchas gracias sería algo asi

```
"exec" : {
        "preload" : {
            "src" : "
function run () { 
 const value = context['property#channel'].find(property => property.propertyType === 'Nombre-Tipo-Propiedad');
  commands.push({ 
    cmd: 'SET_RESPONSES', 
    value: value
  });
  return commands;
}
```
        ", 
            "context" : "property#user,user#me"
        }
    }

**Reply by acedward** [reply]
En el contexto tienes que agregar property#channel tambien

**Reply by NicolasPLG** [reply]
tienes razón lo agregaré muchas gracias @acedward

**Reply by NicolasPLG** [reply]
hola @acedward 

coloqué el exec, pero no me funcionó :(

dejo el question exec completo
```
 "exec" : {
        "preload" : {
            "src" : "function run () {const commands  = [];const value = context['property#channel'].find(property => property.propertyType === 'sesiones');commands.push({cmd: 'SET_RESPONSES',value: value});  commands.push(  {  cmd: 'SET_READONLY',    value: 'true' });return commands}", 
            "context" : "property#channel,user#me"
        }
    }
```
me lo puedes revisar? =>=>?

**Reply by NicolasPLG** [reply]
agregando no sé si afecta que el channel tenga 
```
    "propertyIds" : [
        ObjectId("57b62035e890a1a47bf37110")
    ]
```

**Reply by NicolasPLG** [reply]
hola @acedward traté de hacer cambios pero sin efecto alguno :(. ya que en channel están los propertyIds y pense que en el find debiese estar el nombre de la coleccion que sería properties pero no tuve efecto alguno xD
```
    "exec" : {
        "preload" : {
            "src" : "function run () {const commands  = [];const value = context['property#channel'].find(propertIds => properties.propertyType === 'sesiones');commands.push({cmd: 'SET_RESPONSES',value: value});  commands.push(  {  cmd: 'SET_READONLY',    value: 'true' });return commands}", 
            "context" : "property#channel,channel#self"
        }
    }
```
Edit: Una solución muy rápida aunque no sé cuál será mi impacto cuando actualice la propertyType (en sentido que me respetaría el orden de las propertyTypes)

```
    "exec" : {
        "preload" : {
            "src" : "function run () { const commands = []; const channel = context['channel#self'];commands.push({ cmd: 'SET_RESPONSES', value: channel.propertyIds[0] });return commands;}", 
            "context" : "channel#self,property#channel"
        }
    }
```

Pero el resultado me dió lo que necesitaba.
![image](https://user-images.githubusercontent.com/61419471/135325813-0d168000-b6cf-4362-b4fa-3b789c42bf02.png)
